eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -698,6 +716,155 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder, val
           goto(NORMAL) using d.copy(channelUpdate = channelUpdate1) storing()
       }
 
+    case Event(cmd: CMD_SPLICE, d: DATA_NORMAL) =>
+      val replyTo = if (cmd.replyTo == ActorRef.noSender) sender() else cmd.replyTo
+      d.spliceStatus match {
+        case SpliceStatus.NoSplice =>
+          val parentCommitments = d.metaCommitments.main
+          if (parentCommitments.hasNoPendingHtlcsOrFeeUpdate && parentCommitments.channelFeatures.hasFeature(DualFunding)) {
+            log.info(s"initiating splice with local.in.amount=${cmd.additionalLocalFunding} local.in.push=${cmd.pushAmount} out.amount=${cmd.spliceOut_opt.map(_.amount).sum}")
+            val txInitRbf = TxInitRbf(d.channelId,
+              lockTime = nodeParams.currentBlockHeight.toLong,
+              feerate = nodeParams.onChainFeeConf.feeEstimator.getFeeratePerKw(target = nodeParams.onChainFeeConf.feeTargets.fundingBlockTarget),
+              fundingContribution = cmd.additionalLocalFunding,
+              pushAmount = cmd.pushAmount,
+              spliceOut_opt = cmd.spliceOut_opt)
+            stay() using d.copy(spliceStatus = SpliceStatus.SpliceRequested(cmd.copy(replyTo = replyTo), txInitRbf)) sending txInitRbf
+          } else {
+            log.warning("cannot do splice")
+            replyTo ! Status.Failure(CommandUnavailableInThisState(d.channelId, "CMD_SPLICE_IN", NORMAL))
+            stay()
+          }
+        case _ =>
+          log.warning("cannot initiate splice, another one is already in progress")
+          replyTo ! Status.Failure(InvalidRbfAlreadyInProgress(d.channelId))
+          stay()
+      }
+
+    case Event(msg: TxInitRbf, d: DATA_NORMAL) =>
```

Comment:
We must be very careful and make sure we reject that message unless very explicitly enabled for specific nodes, we don't want random public nodes to start triggering the splice prototype code on our node as we know it has some issues.